### PR TITLE
Finish 49, 50 and 51

### PR DESCRIPTION
**Pull request by meghanstovall**
Finish 49, 50 and 51

### CODE REVIEW

File: app/controllers/admin/items_controller.rb [hunk 1]
```diff
@@ -4,7 +4,13 @@ def index
   end
 
   def update
-    update_item
+    if params[:order_id] != nil
```
Comment (meghanstovall) [inline app/controllers/admin/items_controller.rb:7]: Add conditional for when a merchant wants to fulfill just their items in an order

File: app/controllers/admin/items_controller.rb [hunk 2]
```diff
@@ -80,4 +86,10 @@ def destroy_flash_and_redirect(item)
     redirect_to "/admin/merchants/#{params[:merchant_id]}/items"
   end
 
+  def fulfill_update_inventory(item, order)
+    item.fulfill_item(order)
+    item.update_inventory(order)
+    flash[:notice] = "You have fulfilled the order for #{item.name}"
+    redirect_to "/admin/merchants/#{item.merchant.id}/orders/#{order.id}"
```
Comment (meghanstovall) [inline app/controllers/admin/items_controller.rb:93]: Add method for refactor purposes that calls two other item model methods, shows a flash message and redirects

File: app/controllers/admin/orders_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Admin::OrdersController < Admin::BaseController
+
+  def show
+    @merchant = Merchant.find(params[:merchant_id])
+    @order = Order.find(params[:order_id])
+  end
+end
```
Comment (meghanstovall) [inline app/controllers/admin/orders_controller.rb:7]: Create Admin Orders controller and the show action

File: app/models/item.rb [hunk 4]
```diff
@@ -47,4 +47,18 @@ def updates_active
       update(active?: true)
     end
   end
+
+  def fulfill_item(order)
+    item_orders.each do |item_order|
+      if item_order.order == order
+        item_order.update(status: "fulfilled")
+      end
```
Comment (meghanstovall) [inline app/models/item.rb:55]: Add item model method that will fulfill only items in an order for a certain item and order

File: app/models/item.rb [hunk 5]
```diff
@@ -47,4 +47,18 @@ def updates_active
...
+    end
+  end
+
+  def update_inventory(order)
+    item_order = order.item_orders.where(item: self)
+    new_amount = inventory - item_order.first.quantity
+    update(inventory: new_amount)
+  end
```
Comment (meghanstovall) [inline app/models/item.rb:63]: Add item model method that will update the items inventory when the item is fulfilled

File: app/views/admin/orders/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+        <% if item_order.quantity > item_order.item.inventory %>
+          <h5>Cannot fulfill this item: Not enough inventory</h5>
+        <% else %>
+          <h5><%= link_to "Fulfill Item", "/admin/merchants/#{@merchant.id}/orders/#{@order.id}/items/#{item_order.item.id}", method: :patch %></h5>
+        <% end %>
+      <% end %>
+    </section>
+  <% end %>
+<% end %>
```
Comment (meghanstovall) [inline app/views/admin/orders/show.html.erb:22]: Add an Admin orders show page

File: app/views/merchant/dashboard/show.html.erb [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
  <% if current_user.merchant.item_orders != [] %>
    <% current_user.merchant.item_orders.each do |item_order| %>
-     <h5>Order ID: <%= link_to item_order.order.id, "/orders/:id" %></h5>
+     <h5>Order ID: <%= link_to item_order.order.id, "/orders/#{item_order.order.id}" %></h5>
      <h5>Order Created at: <%= item_order.created_at.strftime("%Y-%m-%d") %></h5>
```
Comment (meghanstovall) [inline app/views/merchant/dashboard/show.html.erb:8]: Made the route for the link to pass through an actual order id

File: config/routes.rb [hunk 8]
```diff
@@ -57,8 +57,10 @@
     patch '/orders/:id', to: "dashboard#update"
     patch '/merchants/:id', to: "merchants#update"
     get '/merchants/:merchant_id', to: "dashboard#show"
+    get '/merchants/:merchant_id/orders/:order_id', to: "orders#show"
     get '/merchants/:merchant_id/items', to: "items#index"
     patch '/merchants/:merchant_id/items/:item_id', to: "items#update"
+    patch '/merchants/:merchant_id/orders/:order_id/items/:item_id', to: "items#update"
     delete '/merchants/:merchant_id/items/:item_id', to: "items#destroy"
     get '/merchants/:merchant_id/items/new', to: "items#new"
```
Comment (meghanstovall) [inline config/routes.rb:65]: Add two routes for the admin order show page and the item update for a certain order

File: spec/features/admins/orders/show_spec.rb [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    expect(page).to have_content("Item: #{@tire.name}")
+    expect(page).to have_content("Price: #{@tire.price}")
+    expect(page).to have_content("Quantity of Items in Order: #{@tire_order.quantity}")
+
+    expect(page).to_not have_content(@paper.name)
+    expect(page).to_not have_content("Quantity of Items in Order: #{@paper_order.quantity}")
+
+    click_link "#{@tire.name}"
+    expect(current_path).to eq("/items/#{@tire.id}")
+  end
```
Comment (meghanstovall) [inline spec/features/admins/orders/show_spec.rb:49]: Test that a merchant user can go to the orders show page and only see items of that merchant

File: spec/features/admins/orders/show_spec.rb [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  it "can fulfill their items in te order" do
+    visit "/admin/merchants/#{@meg.id}/orders/#{@user_order.id}"
+
+    within "#item-#{@tire.id}" do
+      click_link "Fulfill Item"
+    end
+    expect(current_path).to eq("/admin/merchants/#{@meg.id}/orders/#{@user_order.id}")
+    expect(page).to have_content("Status: fulfilled")
+    expect(page).to have_content("You have fulfilled the order for #{@tire.name}")
+  end
```
Comment (meghanstovall) [inline spec/features/admins/orders/show_spec.rb:60]: Test that a merchant user can fulfill only their items in an order

File: spec/features/admins/orders/show_spec.rb [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+  it "cant see fulfill button if quantity is greater than inventory" do
+    visit "/admin/merchants/#{@mike.id}/orders/#{@user_order.id}"
+
+    within "#item-#{@paper.id}" do
+      expect(page).to_not have_link("Fulfill Item")
+      expect(page).to have_content("Cannot fulfill this item: Not enough inventory")
+    end
+  end
```
Comment (meghanstovall) [inline spec/features/admins/orders/show_spec.rb:69]: Test that an item cant be fulfill if there is not enough of the item in merchant's inventory

File: spec/models/item_spec.rb [hunk 12]
```diff
@@ -63,6 +63,28 @@
...
+    it "fulfill_item" do
+      user = User.create(name: "Mike",street_address: "456 Logan St. Denver, CO",
+                                city: "denver",state: "CO",zip: "80206",email: "new_email1@gmail.com",password: "hamburger1", role: 1)
+      order = user.orders.create(name: user.name, address: user.street_address, city: user.city, state: user.state, zip: user.zip)
+      tire = @bike_shop.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
+      tire_order = ItemOrder.create!(item: tire, order: order, price: tire.price, quantity: 7)
+
+      tire.fulfill_item(order)
+      expect(tire.item_orders.first.status).to eq("fulfilled")
+    end
```
Comment (meghanstovall) [inline spec/models/item_spec.rb:76]: Test item model method that will fulfill an item in a certain order

File: spec/models/item_spec.rb [hunk 13]
```diff
@@ -63,6 +63,28 @@
...
+    it "update_inventory" do
+      user = User.create(name: "Mike",street_address: "456 Logan St. Denver, CO",
+                                city: "denver",state: "CO",zip: "80206",email: "new_email1@gmail.com",password: "hamburger1", role: 1)
+      order = user.orders.create(name: user.name, address: user.street_address, city: user.city, state: user.state, zip: user.zip)
+      tire = @bike_shop.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
+      tire_order = ItemOrder.create!(item: tire, order: order, price: tire.price, quantity: 7)
+
+      tire.update_inventory(order)
+      expect(tire.inventory).to eq(5)
+    end
```
Comment (meghanstovall) [inline spec/models/item_spec.rb:87]: Test item model method that will change the item_order status fulfill when a merchants items has been fulfilled

**Comment by palworth** [comment]
Ready for master.

**Pull request merged**